### PR TITLE
save coredumps from cardano-sl

### PR DESCRIPTION
**Pull request by cleverca22**

### CODE REVIEW

File: modules/cardano-node.nix [hunk 1]
```diff
@@ -187,6 +187,7 @@ in {
         key = keyId + ".sk";
       in ''
         [ -f /run/keys/${keyId} ] && cp /run/keys/${keyId} ${stateDir}${key}
+        ulimit -c unlimited
```
Comment (domenkozar) [inline modules/cardano-node.nix]: Why is this needed? Would be good to have a comment. Is it save to have it on for production?

File: modules/common.nix [hunk 2]
```diff
@@ -24,6 +24,8 @@ in {
 
   environment.variables.TERM = "xterm-256color";
 
+  systemd.coredump.enable = true;
```
Comment (domenkozar) [inline modules/common.nix]: Is it safe to have coredump happen in production?

File: modules/cardano-node.nix (hunk 1 above)
Comment (cleverca22) [inline modules/cardano-node.nix]: systemd.coredump.enable only sets a hook that captures cores, ulimit is still required to enable any given program to dump

File: modules/common.nix (hunk 2 above)
Comment (cleverca22) [inline modules/common.nix]: hmmm, it will contain copies of the private key data in the cores

**Comment by domenkozar** [comment]
@cleverca22 sadly `productionMode` is a different thing than the production environment. I think we want to enable these only for staging environment (which are the files suffixed with -env-staging.nix`
